src/test/java/io/lettuce/core/internal/AsyncConnectionProviderTest.java

```diff
@@ -0,0 +1,99 @@
+package io.lettuce.core.internal;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class AsyncConnectionProviderTest {
+
+    @Test
+    public void testFutureListLength() throws InterruptedException, ExecutionException, TimeoutException {
+
+        CountDownLatch slowCreate = new CountDownLatch(1);
+        CountDownLatch slowShutdown = new CountDownLatch(1);
+
+        // create a provider with a slow connection creation
```

Comment:
that's means it will create an `Sync` but hasn't completed yet, even when we need to close provider